cmd/config/docs/commands/upgrade.md

```diff
@@ -0,0 +1,19 @@
+## set
+
+Upgrade setters for Resources.
+
+### Synopsis
+
+Upgrade setters for Resources.
```

Comment:
You might want to make this help less specific, i.e. not call out _setters_ specifically.

kustomize has an `edit fix` command that reads in a kustomization file, purports to looks for deprecated stuff, converts it to whatever the new format should be, and writes it out.  It does more than just deal with deprecations - fields will be sorted, indented, etc. (like `go fmt`).